### PR TITLE
auth0: Expose more of the Job resource in management client

### PR DESCRIPTION
**Pull request by pacey**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://auth0.com/docs/api/management/v2#!/Jobs/get_jobs_by_id
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by pacey** [comment]
The version in the definition header seems to have fallen out of sync. There is 2.9.3 in Git, but 2.9.10 is published to NPM. I can update this to 2.9.11 if required?

File: types/auth0/index.d.ts [hunk 1]
```diff
@@ -509,13 +509,64 @@ export interface StatsParams {
   to: string;
 }
 
+export type Job = ImportUsersJob | ExportUsersJob | VerificationEmailJob;
+
+export type JobFormat = 'csv' | 'json';
+
+export type JobStatus = 'pending' | 'completed';
```
Comment (pacey) [inline types/auth0/index.d.ts]: I would assume there is another status for "running" or "in_progress", but I've never seen it and I can't find anything in the Auth0 docs to suggest it exists/what the value is

Comment (pacey) [reply types/auth0/index.d.ts]: I found "processing"and "failed" mentioned here https://community.auth0.com/t/management-api-we-need-better-docs/18910/2

**Comment by armanio123** [comment]
> The version in the definition header seems to have fallen out of sync. There is 2.9.3 in Git, but 2.9.10 is published to NPM. I can update this to 2.9.11 if required?

DefinitelyTyped uses the patch version itself. It will auto-increment the patch version for every update of the same minor version, so it's impossible to have types that target only a patch version. They must work for all patches of a minor version.

File: types/auth0/index.d.ts [hunk 2]
```diff
@@ -509,13 +509,64 @@ export interface StatsParams {
...
+
 export interface ImportUsersOptions {
-  connection_id: string;
-  users: string;
+    users: string;
+    connection_id: string;
+    upsert?: boolean;
+    external_id?: string;
+    send_completion_email?: boolean;
 }
```
Comment (armanio123) [inline types/auth0/index.d.ts]: Changing the interface name from `UserIdParams` to `VerificationEmailOptions` will break users using this type. I don't think that's an intended behavior. I suggest changing back the name unless you feel this is a different interface that needs to be fixed.

**Review by armanio123 — CHANGES_REQUESTED** [review]

**Comment by pacey** [comment]
Thanks for having a look @armanio123, I'll have a look when I get into work

File: types/auth0/index.d.ts (hunk 2 above)
Comment (pacey) [reply types/auth0/index.d.ts]: The name change has been reverted

**Review by armanio123 — APPROVED** [review]

**Pull request merged**